stochss/handlers/util/parameter_sweep_2d.py

```diff
@@ -140,32 +141,74 @@ def get_plotly_traces(self, keys):
+    #     divider = mpl_toolkits.axes_grid1.make_axes_locatable(axis)
+    #     cax = divider.append_axes("right", size="5%", pad=0.2)
+    #     _ = matplotlib.pyplot.colorbar(ax=axis, cax=cax)
+
+
+    @classmethod
+    def plot(cls, results, species, params, mapper="final", reducer="avg"):
         '''
-        Plot the results based on the keys using matplotlib
+        Plot the results with error bar from time series results.
 
         Attributes
         ----------
-        key : list
-            Identifiers for the results data
+        results : list
+            List of GillesPy2 results objects.
+        species : str
+            Species of interest name.
+        params : list
+            List of StochSS sweep parameter dictionaries.
+        mapper : str
+            Key indicating the feature extraction function to use.
+        reducer : str
+            Key indicating the ensemble aggragation function to use.
         '''
-        if len(keys) <= 2:
-            results = self.results[keys[0]][keys[1]]
-        else:
-            results = self.results[keys[0]][keys[1]][keys[2]]
-
-        _, axis = matplotlib.pyplot.subplots(figsize=(8, 8))
-        matplotlib.pyplot.imshow(results)
-        axis.set_xticks(numpy.arange(results.shape[1])+0.5, minor=False)
-        axis.set_yticks(numpy.arange(results.shape[0])+0.5, minor=False)
-        matplotlib.pyplot.title(f"Parameter Sweep - Variable: {keys[0]}")
-        axis.set_xticklabels(self.params[0]['range'], minor=False, rotation=90)
-        axis.set_yticklabels(self.params[1]['range'], minor=False)
-        axis.set_xlabel(self.params[0]['parameter'], fontsize=16, fontweight='bold')
-        axis.set_ylabel(self.params[1]['parameter'], fontsize=16, fontweight='bold')
-        divider = mpl_toolkits.axes_grid1.make_axes_locatable(axis)
-        cax = divider.append_axes("right", size="5%", pad=0.2)
-        _ = matplotlib.pyplot.colorbar(ax=axis, cax=cax)
+        func_map = {"min": numpy.min, "max": numpy.max, "avg": numpy.mean,
+                    "var": numpy.var, "final": lambda res: res[-1]}
+        data = []
+        for p_results in results:
+            map_results = [[func_map[mapper](traj[species]) for traj in result]
+                            for result in p_results]
+            if len(map_results[0]) > 1:
+                red_results = [func_map[reducer](map_result) for map_result in map_results]
+            else:
+                red_results = [map_result[0] for map_result in map_results]
+            data.append(red_results)
+        data = numpy.array(data)
+
+        trace_list = [plotly.graph_objs.Heatmap(z=data, x=params[0]['range'],
+                                                y=params[1]['range'])]
+
+        title = f"<b>Parameter Sweep - Variable: {species}</b>"
+        layout = plotly.graph_objs.Layout(title=dict(text=title, x=0.5),
+                                          xaxis=dict(title=f"<b>{params[0]['name']}</b>"),
```

Comment:
This is a XSS vulnerability, and allows for arbitrary code execution on published models.

To replicate: set your species name to a JavaScript code segment, like `<script>window.alert("ping!")</script>` and run a parameter sweep simulation. It's validated on the front end, but not on the server.

![image](https://user-images.githubusercontent.com/42880781/127943507-deb63779-8162-4363-a793-d9a33c41aca7.png)